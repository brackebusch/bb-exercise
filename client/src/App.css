.App {
  background-color: #fff;
  height: 1125px;
  width: 1025px;
  border: 1px solid black;
  color: rgb(85, 85, 85);
  margin: auto;
  margin-top: 210px;
  /* display: flex;
  text-align: center;
  flex-direction: column;
  align-items: center;
  justify-content: center; */
}

.Body {
  width: 960px;
  margin: auto;
  display: flex;
  flex-direction: column;
}

button {
  font-size: 12px;
  letter-spacing: 1.5px;
  height: 40px;
  width: 118px;
  text-transform: uppercase;
  transition: all .3s;
  color: #fff;
  background-color: #01a1dd;
  border: none;
  box-shadow: none;
  font-weight: 400;
  outline: none;
  text-shadow: none;
  padding: 1px 15px;
  text-align: center;
}

button:hover {
  background-color: #0088cc;
  cursor: pointer;
}

img {
  height: 32px;
  width: 225px;
  align-self: flex-start;
  margin: 15px 0px;
}

.Header {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 29px;
}

.Date-Title {
  display: flex;
  align-items: center;
  justify-content: center;
}

.Title {
  font-weight: 400;
  font-size: 32px;
  line-height: 32px;
  text-size-adjust:100%;
  text-rendering:optimizeLegibility;
  margin-left: 32px;
}

.Date {
  background-color: #f3f3f3;
  height: 70px;
  width: 70px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  border-radius: 3px;
} 

.Month {
  text-transform: uppercase;
  font-weight: 700;
  letter-spacing: 1.5px;
  font-size: 14px;
  text-align: center;
}

.Day {
  font-size: 28px;
  letter-spacing: 2px;
  text-align: center;
}

.Table {
  border: 2px solid rgb(243, 243, 243);
}

.Table h4 {
  color: rgb(1, 161, 221);
  font-weight: 800;
  font-size: 13px;
  margin: 24px 0px 10px 37px;
  letter-spacing: 1.5px;
}

hr {
  border: .5px solid  rgb(229, 229, 229);
  width: 920px;
  margin: 0 20px;
}

table {
  font-size: 12px;
}