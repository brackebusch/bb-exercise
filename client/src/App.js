import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import moment from 'moment';

class App extends Component {
  state = {data: null}

  async componentDidMount(){
    const response = await fetch('/work_orders')
    const data = await response.json()
    this.setState({data})
    console.log(data)
  }

  render() {
    return (
      <div className='App'>
        <div className='Body'>
          <img src={logo} className='App-logo' alt='logo' />

          <div className='Header'>
            <div className='Date-Title'>
              <div className='Date'>
                <span className='Month'>{moment().format('MMM')}</span>
                <span className='Day'>{moment().format('DD')}</span>
              </div>
              <span className='Title'>Perfectly Ground Work Work Orders</span>
            </div>
            <button>CREATE ORDER</button>
          </div>

          <div className='Table'>
            <h4>ORDERS</h4>
            <hr/>
            <table>
              <tr>
                <th>Coffee</th>
                <th>Method</th>
                <th>Number of Cases</th>
                <th>Packets per Case</th>
                <th>Ship Date</th>
                <th>Order</th>
                <th>View</th>
              </tr>
            </table>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
