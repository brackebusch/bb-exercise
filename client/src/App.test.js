import React from 'react';
import App from './App';
import moment from 'moment'

it('renders without crashing', () => {
  const wrapper = shallow(
    <App />
  );

  expect(wrapper).toMatchSnapshot()
});

it('set state when date is passed to setDate', () => {
  let date = moment()
  const wrapper = mount(<App />);
  const instance = wrapper.instance();
  expect(wrapper.state('workOrder').ship_date).toBe('');
  instance.setDate(date);
  expect(wrapper.state('workOrder').ship_date).toBe(date);
});