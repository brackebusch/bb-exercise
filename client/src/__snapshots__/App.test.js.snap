// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="Body"
      >
        <Header
          changeModalState={[Function]}
        />
        <ModalView
          changeModalState={[Function]}
          error=""
          handleInputChange={[Function]}
          saveAndClose={[Function]}
          setDate={[Function]}
          setError={[Function]}
          showModal={false}
          workOrder={
            Object {
              "cases": "#",
              "coffee": "Choose One",
              "method": "Choose One",
              "notes": "",
              "packets": "25",
              "priority": false,
              "ship_date": "",
            }
          }
        />
        <Table
          changeModalState={[Function]}
          reverse={[Function]}
          workOrders={Array []}
        />
        <Pagination
          bsClass="pagination"
          bsSize="medium"
        >
          <First />
          <PaginationItem
            active={false}
            activeLabel="(current)"
            disabled={false}
            onClick={[Function]}
          >
            Prev
          </PaginationItem>
          <PaginationItem
            active={false}
            activeLabel="(current)"
            disabled={false}
            onClick={[Function]}
          >
            Next
          </PaginationItem>
          <Last />
        </Pagination>
      </div>,
      "className": "App",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            changeModalState={[Function]}
          />,
          <ModalView
            changeModalState={[Function]}
            error=""
            handleInputChange={[Function]}
            saveAndClose={[Function]}
            setDate={[Function]}
            setError={[Function]}
            showModal={false}
            workOrder={
              Object {
                "cases": "#",
                "coffee": "Choose One",
                "method": "Choose One",
                "notes": "",
                "packets": "25",
                "priority": false,
                "ship_date": "",
              }
            }
          />,
          <Table
            changeModalState={[Function]}
            reverse={[Function]}
            workOrders={Array []}
          />,
          <Pagination
            bsClass="pagination"
            bsSize="medium"
          >
            <First />
            <PaginationItem
              active={false}
              activeLabel="(current)"
              disabled={false}
              onClick={[Function]}
            >
              Prev
            </PaginationItem>
            <PaginationItem
              active={false}
              activeLabel="(current)"
              disabled={false}
              onClick={[Function]}
            >
              Next
            </PaginationItem>
            <Last />
          </Pagination>,
        ],
        "className": "Body",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeModalState": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeModalState": [Function],
            "error": "",
            "handleInputChange": [Function],
            "saveAndClose": [Function],
            "setDate": [Function],
            "setError": [Function],
            "showModal": false,
            "workOrder": Object {
              "cases": "#",
              "coffee": "Choose One",
              "method": "Choose One",
              "notes": "",
              "packets": "25",
              "priority": false,
              "ship_date": "",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "changeModalState": [Function],
            "reverse": [Function],
            "workOrders": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "pagination",
            "bsSize": "medium",
            "children": Array [
              <First />,
              <PaginationItem
                active={false}
                activeLabel="(current)"
                disabled={false}
                onClick={[Function]}
              >
                Prev
              </PaginationItem>,
              Array [],
              <PaginationItem
                active={false}
                activeLabel="(current)"
                disabled={false}
                onClick={[Function]}
              >
                Next
              </PaginationItem>,
              <Last />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "active": false,
                "activeLabel": "(current)",
                "children": "Prev",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Prev",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "active": false,
                "activeLabel": "(current)",
                "children": "Next",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Next",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Body"
        >
          <Header
            changeModalState={[Function]}
          />
          <ModalView
            changeModalState={[Function]}
            error=""
            handleInputChange={[Function]}
            saveAndClose={[Function]}
            setDate={[Function]}
            setError={[Function]}
            showModal={false}
            workOrder={
              Object {
                "cases": "#",
                "coffee": "Choose One",
                "method": "Choose One",
                "notes": "",
                "packets": "25",
                "priority": false,
                "ship_date": "",
              }
            }
          />
          <Table
            changeModalState={[Function]}
            reverse={[Function]}
            workOrders={Array []}
          />
          <Pagination
            bsClass="pagination"
            bsSize="medium"
          >
            <First />
            <PaginationItem
              active={false}
              activeLabel="(current)"
              disabled={false}
              onClick={[Function]}
            >
              Prev
            </PaginationItem>
            <PaginationItem
              active={false}
              activeLabel="(current)"
              disabled={false}
              onClick={[Function]}
            >
              Next
            </PaginationItem>
            <Last />
          </Pagination>
        </div>,
        "className": "App",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header
              changeModalState={[Function]}
            />,
            <ModalView
              changeModalState={[Function]}
              error=""
              handleInputChange={[Function]}
              saveAndClose={[Function]}
              setDate={[Function]}
              setError={[Function]}
              showModal={false}
              workOrder={
                Object {
                  "cases": "#",
                  "coffee": "Choose One",
                  "method": "Choose One",
                  "notes": "",
                  "packets": "25",
                  "priority": false,
                  "ship_date": "",
                }
              }
            />,
            <Table
              changeModalState={[Function]}
              reverse={[Function]}
              workOrders={Array []}
            />,
            <Pagination
              bsClass="pagination"
              bsSize="medium"
            >
              <First />
              <PaginationItem
                active={false}
                activeLabel="(current)"
                disabled={false}
                onClick={[Function]}
              >
                Prev
              </PaginationItem>
              <PaginationItem
                active={false}
                activeLabel="(current)"
                disabled={false}
                onClick={[Function]}
              >
                Next
              </PaginationItem>
              <Last />
            </Pagination>,
          ],
          "className": "Body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "changeModalState": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "changeModalState": [Function],
              "error": "",
              "handleInputChange": [Function],
              "saveAndClose": [Function],
              "setDate": [Function],
              "setError": [Function],
              "showModal": false,
              "workOrder": Object {
                "cases": "#",
                "coffee": "Choose One",
                "method": "Choose One",
                "notes": "",
                "packets": "25",
                "priority": false,
                "ship_date": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "changeModalState": [Function],
              "reverse": [Function],
              "workOrders": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "pagination",
              "bsSize": "medium",
              "children": Array [
                <First />,
                <PaginationItem
                  active={false}
                  activeLabel="(current)"
                  disabled={false}
                  onClick={[Function]}
                >
                  Prev
                </PaginationItem>,
                Array [],
                <PaginationItem
                  active={false}
                  activeLabel="(current)"
                  disabled={false}
                  onClick={[Function]}
                >
                  Next
                </PaginationItem>,
                <Last />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "active": false,
                  "activeLabel": "(current)",
                  "children": "Prev",
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Prev",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "active": false,
                  "activeLabel": "(current)",
                  "children": "Next",
                  "disabled": false,
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Next",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
